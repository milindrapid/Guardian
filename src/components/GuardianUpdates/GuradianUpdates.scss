$main-title-color: #0d3f5e;
$link-color: #00728d;
$border-color: #a7a8aa;

.base-container {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  border: 1px solid $border-color;
  background-color: white;
  padding-left: 1.5rem;
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
  margin: 2rem 2rem 0 2rem;
}
.title-container {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}
.update-main-title {
  font-size: x-large;
  font-weight: bold;
  color: $main-title-color;
  margin-bottom: 10px;
}
.move-icon {
  width: 20px;
  height: 20px;
  margin-left: 10px;
  margin-right: 10px;
}
.more-icon {
  width: 25px;
  height: 25px;
  margin-left: 10px;
  margin-right: 10px;
}
.item-container {
  padding-left: 0;
}
.data-content {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  position: relative;
}

.item-title {
  font-size: large;
  color: #04012e;
  font-weight: bold;
  margin-top: 1rem;
}
.item-link {
  width: fit-content;
  font-size: medium;
  color: $link-color;
  font-weight: bold;
  text-decoration: underline;
  cursor: pointer;
  margin-top: 5px;
  margin-bottom: 5px;
}
.item-content {
  width: fit-content;
  font-size: medium;
  color: black;
  font-weight: normal;
}

.close-image {
  right: 0;
  font-size: 16px;
  font-weight: 900;
  padding: 0.75rem;
  margin-right: 25px;
  margin-top: 5px;
  cursor: pointer;
  position: absolute;
}

.hr-line {
  width: 100%;
  height: 1px;
  background-color: grey;
  margin-top: 1rem;
  margin-bottom: 1rem;
  opacity: 0.3;
}
.go-to-link-container {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}
.go-to-link {
  width: fit-content;
  font-size: medium;
  font-weight: bold;
  color: $link-color;
  text-decoration: underline;
}
.go-to-link-icon {
  width: 15px;
  height: 20px;
  align-self: center;
  margin-left: 0.5rem;
}
