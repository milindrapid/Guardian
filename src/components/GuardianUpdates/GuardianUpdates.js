import React from 'react';
import './GuradianUpdates.scss';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const GuradianUpdatesBaseContainer = styled.div`
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
    border: 1px solid #A7A8AA;
    background-color: white;
    padding-left: 1.5rem;
    padding-top: 1.5rem;
    padding-bottom: 1.5rem;
    margin: 2rem 2rem 0 2rem;

    mainTitle{
        font-style:normal;
        font-size: 1.5rem;
        line-height:1.625rem;
        font-weight: bold;
        color: #0D3F5E;
        margin-bottom: .625rem;
    }

    ul{
        padding-left: 0;
    }
`;

const DataContent = styled.div`
        display: flex;
        flex-direction: column;
        flex-wrap: wrap;
        position: relative;

        closeImage{
            right: 0;
            font-size: 1rem;
            font-weight: 900;
            padding: 0.75rem;
            margin-right: 1.5rem;
            margin-top: .3rem;
            cursor: pointer;
            position: absolute;
        }

        h1{
            color: #2B2B2B;
            margin-top: 1rem;
            font-style:normal;
            font-weight: bold;
            font-size: 1.25rem;
            line-height:1.5rem;
        }

        a{
            width: fit-content;
            font-size: 1rem;
            line-height:1.25rem;
            color: #00728d;
            font-style:normal;
            font-weight: bold;
            text-decoration: underline;
            cursor: pointer;
            margin-bottom:.5rem;
            &:hover{
                color: #00728d;
            }
        }
        span{
            width: fit-content;
            font-style:normal;
            font-weight: normal;
            font-size: 1rem;
            line-height:1.3rem;
        }
        horizontelLine{
            width: 100%;
            height: 1px;
            background-color: grey;
            margin-top: 1rem;
            margin-bottom: 1rem;
            opacity: 0.3;
        }
`;

const GotoLinkContainer = styled.div`
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;

    a{
        width: fit-content;
        font-size: 1rem;
        color: #00728d;
        text-decoration:underline;
        font-style:normal;
        font-weight: bold;
        line-height:1.25rem;
    }

    rightArrow{
        font-style:normal;
        font-weight:900;
        font-size:1rem;
        line-height:1rem;
        align-self: center;
        color:#0D3F5E;
        margin-left: 0.5rem;
    }
`;

const GuradianUpdates = (props) => {

    const { data } = props;

    const getGuardianUpdates = () => {
        return (
            data.map(update => (
                <DataContent>
                    <closeImage><FontAwesomeIcon icon='times' /></closeImage>
                    <h1>{update.title}</h1>
                    <a href='/'>{update.link}</a>
                    <span>{update.content}</span>
                    <horizontelLine />
                </DataContent>
            ))
        )
    }

    return (
        <GuradianUpdatesBaseContainer>
            <mainTitle>Updates from Guardian</mainTitle>
            <ul>{getGuardianUpdates()}</ul>
            <GotoLinkContainer>
                <a href='/'>Go to Resource Center</a>
                <rightArrow><FontAwesomeIcon icon='arrow-right' /></rightArrow>
            </GotoLinkContainer>
        </GuradianUpdatesBaseContainer>
    )
}

GuradianUpdates.propTypes = {
    data: PropTypes.array.isRequired
}

export default GuradianUpdates;