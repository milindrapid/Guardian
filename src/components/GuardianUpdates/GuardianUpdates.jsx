import React from 'react';
import './GuradianUpdates.scss';

const GuradianUpdates = (props) => {

    const onItemClick = (link) => {
        props.onItemClick(link)
        // alert(link)
    }

    const getGuardianUpdates = () => {
        return (
            props.data.map(update => (
                <div className="data-content">
                    <img className="close-image" src={require('../../images/close.svg')} alt="close" />
                    <label className="item-title">{update.title}</label>
                    <label className="item-link" onClick={() => onItemClick(update.link)}>{update.link}</label>
                    <label className="item-content">{update.content}</label>
                    <div className="hr-line" />
                </div>
            ))
        )
    }

    return (
        <div className="base-container">
            <span className="update-main-title">Updates from Guardian</span>
            <div>
                <ul className="item-container">
                    {getGuardianUpdates()}
                </ul>
            </div>
            <div className="go-to-link-container">
                <a className="go-to-link" href="/">Go to Resource Center</a>
                <img className="go-to-link-icon" src={require('../../images/right-arrow.svg')} alt="close" />
            </div>
        </div>
    )
}
export default GuradianUpdates;