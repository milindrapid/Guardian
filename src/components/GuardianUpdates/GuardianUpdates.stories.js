import React from "react";
import GuardianUpdates from "./GuardianUpdates";

const updatesFromGuardian = [
    { id: 1, title: "Marketing", link: 'Coronavirus and Market Volatility Playbook', content: 'Your resource of managing coronavirus and market volatility' },
    { id: 2, title: "New Feature", link: 'Financial Managemen Tool', content: 'Your resource of managing coronavirus and market volatility' },
    { id: 3, title: "Product Update", link: 'Bank Draft cutoff', content: 'Your resource of managing coronavirus and market volatility' },
]

export default {
    title: 'Updates from Guardian',
    component: GuardianUpdates,
};

const props = {
    data: updatesFromGuardian,
};

export const UpdatesFromGuardian = () => <GuardianUpdates {...props} />;

UpdatesFromGuardian.story = {
    name: 'Updates From Guardian'
};