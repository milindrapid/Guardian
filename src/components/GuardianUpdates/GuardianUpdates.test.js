import React from 'react';
import GuardianUpdates from './GuardianUpdates';
import { shallow } from 'enzyme';

const updatesFromGuardian = [
    { id: 1, title: 'Marketing', link: 'Coronavirus and Market Volatility Playbook', content: 'Your resource of managing coronavirus and market volatility' },
]


describe('GuardianUpdates component', () => {
    const props = {
        data: updatesFromGuardian,
    };

    const wrapperComponent = shallow(<GuardianUpdates {...props} />);

    it('Global header component should render correclty', () => {
        expect(wrapperComponent).toMatchSnapshot();
    });

    it('should render children components correclty', () => {
        expect(wrapperComponent.find('ul')).toHaveLength(1);
        expect(wrapperComponent.find('a')).toHaveLength(2);
        expect(wrapperComponent.find('h1')).toHaveLength(1);
        expect(wrapperComponent.find('span')).toHaveLength(1);
    });
});
