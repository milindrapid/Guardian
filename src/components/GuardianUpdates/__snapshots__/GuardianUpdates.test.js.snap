// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuardianUpdates component Global header component should render correclty 1`] = `
<styled.div>
  <mainTitle>
    Updates from Guardian
  </mainTitle>
  <ul>
    <styled.div>
      <closeImage>
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="times"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </closeImage>
      <h1>
        Marketing
      </h1>
      <a
        href="/"
      >
        Coronavirus and Market Volatility Playbook
      </a>
      <span>
        Your resource of managing coronavirus and market volatility
      </span>
      <horizontelLine />
    </styled.div>
  </ul>
  <styled.div>
    <a
      href="/"
    >
      Go to Resource Center
    </a>
    <rightArrow>
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="arrow-right"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    </rightArrow>
  </styled.div>
</styled.div>
`;
